### PR TITLE
Possible solution for #27

### PR DESCRIPTION
**Pull request by L0RD-ZER0**
A possible solution for solving #27 by using `where` command on windows

### CODE REVIEW

File: doculog/git.py [hunk 1]
```diff
@@ -12,25 +12,16 @@
...
+        git_paths = subprocess.check_output(["where", "git"]).decode(sys.stdout.encoding).split('\n')
+        try:
+            for exepath in git_paths:
+                exepath = exepath.strip()
+                if os.access(exepath, os.X_OK):
+                    return exepath
+        except OSError:
+            pass
+
+            return "git.cmd"
```
Comment (TTitcombe) [inline doculog/git.py:24]: This should be un-nested from `pass`, right?

**Review by TTitcombe — COMMENTED** [review]
Thanks for this PR! I think this should work, but I'm getting reports from some users that this still doesn't work for them. I'm looking into it. meanwhile, we'll keep this open as it might be tweaked into a fully working state.

**Comment by L0RD-ZER0** [comment]
> Thanks for this PR! I think this should work, but I'm getting reports from some users that this still doesn't work for them. I'm looking into it. meanwhile, we'll keep this open as it might be tweaked into a fully working state.

Not having git in path is also a bit of an issue with the current solution.

Would like some insight on the issue being faced by other users so can ponder over the necessary tweaks ~~and maybe come up with a better one while at it~~. Another thing which can be done is to open and read logs from the `.git` folder of the root directly or alternatively bundle git with the lib.

On a side-note, I thought of using a recursive where call with `where /R C:\ git.*` but issue with that could be files made to interract with git but not the git executable itself.

**Comment by TTitcombe** [comment]
The user with a failure case gets 
```
[ "C:\\Program Files (x86)\\Git\\cmd\\git.exe\r", ""]
```
as the result from the `where git` subprocess call, and `subprocess.check_output(["C:\\Program Files (x86)\\Git\\cmd\\git.exe", "log"])` works, so I'm a little confused why this function isn't working for them.

**Comment by L0RD-ZER0** [comment]
Rather interesting

File: doculog/git.py (hunk 1 above)
Comment (L0RD-ZER0) [reply doculog/git.py:24]: Yeah, my bad there.

**Comment by L0RD-ZER0** [comment]
Update: I used an old Win-7 system and on there. Funnily enough, the program itself works and changelog was generated, but pytest errors there. So I fixed it using `Popen` instead of `check_output` and both pytest as well as the whole thing it seems to be working on there now as well. Do review it.

**Comment by TTitcombe** [comment]
Thanks for that @L0RD-ZER0 , looks good. I'll merge in now as it's definitely better than what we currently have, but I'll still waiting on confirmation from the people who claimed to still be seeing an error. We'll be in touch on discord - what's your username there?

**Comment by L0RD-ZER0** [comment]
> Thanks for that @L0RD-ZER0 , looks good. I'll merge in now as it's definitely better than what we currently have, but I'll still waiting on confirmation from the people who claimed to still be seeing an error. We'll be in touch on discord - what's your username there?

You can use `The One#9690` to get in touch. Alternatively, I also use `ℒØℜⅅ ƵℇℜØ#6266` but I mostly use it for gaming and entertainment purposes and not code.

Edit: Didn't expect GH to render tofu characters.

**Pull request merged**